tests/testapp/test_forms.py

```diff
@@ -306,4 +307,37 @@ def test_invalid(self):
     def test_prepare_value(self):
         field = JSONField()
         assert field.prepare_value({'a': 'b'}) == '{"a": "b"}'
+        assert field.prepare_value(['a', 'b']) == '["a", "b"]'
+        assert field.prepare_value(True) == 'true'
+        assert field.prepare_value(False) == 'false'
+        assert field.prepare_value(3.14) == '3.14'
         assert field.prepare_value(None) == 'null'
+        assert field.prepare_value('foo') == '"foo"'
+
+    def test_redisplay_wrong_input(self):
+        """
+        When displaying a bound form (typically due to invalid input), the form
+        should not overquote JSONField inputs.
+        """
+        class JsonForm(forms.Form):
+            name = forms.CharField(max_length=2)
+            jfield = JSONField()
+
+        # JSONField input is fine, name is too long
+        form = JsonForm({'name': 'xyz', 'jfield': '["foo"]'})
+        self.assertIn('[&quot;foo&quot;]</textarea>', form.as_p())
```

Comment:
our tests are on pytest and use plain `assert` everywhere, so this shoudl be `assert '<snip>' in form.as_p()` etc.